### PR TITLE
Add slow_test as it's no longer part of Qiskit 1.0

### PR DESCRIPTION
**Pull request by woodsp-ibm**
### Summary

`slow_test` used to be part of `qiskit.test` but the whole `test` folder was removed for Qiskit 1.0. This adds a copy of that slow_test decorator locally.

Additionally this updates other units testing for failures that started since this was created - see comments below for more detail.

### Details and comments

The Monitor jobs, that run against Qiskit main branch, started failing last night after the removal.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
Since I did this it seems there is a new failure in the 1.0 Monitor dues to the change in  signature of `inverse` method. This adds the additional parameter so lint is ok with it in 1.0, the extra parameter is ok with lint in earlier versions e.g 0.46. Rather than create a separate PR as this was to fix the 1.0 monitoring jobs, and it just again test logic change, I included it here.

**Comment by woodsp-ibm** [comment]
Hmm., now its running tests, which it did not get that far on nightly CI, it seems there are a whole bunch of failures that did not happen a couple of days ago when I first pushed this PR.

**Comment by woodsp-ibm** [comment]
Seems the failures are due Estimator now only supporting SparsePauliOp in 1.0. So where tests had a SparsePauliOp and then done to_matrix() on this to create a matrix op, and tested with both, I dropped the conversion and test using that.

In one place a different operator was created to check re-use, I converted that to a SparsePauliOp.

File: test/time_evolvers/test_pvqd.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class WhatAmI(Gate):
     def __init__(self, angle):
         super().__init__(name="whatami", num_qubits=2, params=[angle])
 
-    def inverse(self):
+    def inverse(self, annotated: bool = False):
```
Comment (1ucian0) [inline test/time_evolvers/test_pvqd.py:51]: This probably deserves a comment that can be removed once Qiskit 0.46 reaches eol.

**Review by 1ucian0 — COMMENTED** [review]
Just a small comment about a comment in `inverse` :)

The rest, LGTM

File: test/time_evolvers/test_pvqd.py (hunk 1 above)
Comment (ElePT) [reply test/time_evolvers/test_pvqd.py:51]: After a quick look at https://github.com/Qiskit/qiskit/pull/11593 it looks like that argument is intended to live in 1.0+, so the comment wouldn't be necessary, right?

**Review by ElePT — COMMENTED** [review]
This looks good to me, maybe we can update the type hints in a quick follow-up PR instead of with #136.

File: test/time_evolvers/test_pvqd.py (hunk 1 above)
Comment (woodsp-ibm) [reply test/time_evolvers/test_pvqd.py:51]: That argument is new in 1.0 - and while adding it now is compatible with 0.46 as it was without causes lint to fail as you can see in the CI jobs which have recently been merged where the monitor jobs failed (they are not required to pass to merge)
```
************* Module test.time_evolvers.test_pvqd
test/time_evolvers/test_pvqd.py:51:4: W0221: Number of parameters was 2 in 'Instruction.inverse' and is now 1 in overriding 'WhatAmI.inverse' method (arguments-differ)
```
And this is just in a test case to not main code

**Comment by woodsp-ibm** [comment]
> This looks good to me, maybe we can update the type hints in a quick follow-up PR 

I don't know whats the right thing here. If you use 0.46 then the hints are correct as the primitives support a wider set. If you use the new 1.0 then its really only then its just SparsePauliOp. Since most people arguably just use SparsePauliOp - the apps all - maybe it does not matter too much. If you had code that used one of the other operators it would still work with 0.46 albeit that it raises a deprecation warning rather than fail as it does if you use 1.0. If we changed to SparsePauliOp only then maybe mypy would complain on your code with the other operators.

**Review by ElePT — APPROVED** [review]
Ok, let's go ahead with this PR and I think we can leave the `BaseOperator` typehint for compatibility with 0.46, I agree that it's not too harmful.

File: test/time_evolvers/test_pvqd.py (hunk 1 above)
Comment (ElePT) [reply test/time_evolvers/test_pvqd.py:51]: Ok, then I think that the comment is not necessary here.

**Pull request merged**